### DISCUSSION TITLE
Automatic renewal

### DISCUSSION THREAD

**Discussion by alimdi** [polymind-inc/acmebot#335] [Q&A] [answered]
Hi,

Thank you for that great application that makes handling certificates easier.
Looks like the automatic renewal isn't performed as some certificates are expiring in less than 2 weeks. Is there a way to debug what's done by the scheduler to identify the cause?

Many thanks for your help.

**Comment by shibayan** [comment, marked as answer]
First of all, please check if the error log remains in the Application Insights deployed with Acmebot.

If you cannot find the error log, check if the RenewCertificates_Orchestrator function is running. If it is not running, restarting Azure Functions may help.

**Reply by alimdi** [reply]
Hi @shibayan,

Thank you for your help.
RenewCertificates_Orchestrator is working fine but the renewal procedure fails on IssueCertificate with a rate limit issue.
That's strange as there are only 5 certificates on the keyvault and not all of them are ready to be renewed. By the way when I tried to renew manually, it worked.

`The orchestrator function 'IssueCertificate' failed: "The activity function 'Order' failed: "Failed to deserialize exception from TaskActivity: {"$type":"ACMESharp.Protocol.AcmeProtocolException, ACMESharp","ProblemType":15,"ProblemTypeRaw":"urn:ietf:params:acme:error:rateLimited","ProblemDetail":"Rate limit for '/acme' reached","ProblemStatus":-1,"StackTrace":"   at ACMESharp.Protocol.AcmeProtocolClient.SendAcmeAsync(Uri uri, HttpMethod method, Object message, HttpStatusCode[] expectedStatuses, Boolean skipNonce, Boolean skipSigning, Boolean includePublicKey, CancellationToken cancel, String opName)\r\n   at ACMESharp.Protocol.AcmeProtocolClient.CreateOrderAsync(IEnumerable 1 identifiers, Nullable 1 notBefore, Nullable 1 notAfter, CancellationToken cancel)\r\n   at KeyVault.Acmebot.Functions.SharedActivity.Order(IReadOnlyList 1 dnsNames) in /home/runner/work/keyvault-acmebot/keyvault-acmebot/KeyVault.Acmebot/Functions/SharedActivity.cs:line 131\r\n   at Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker² 2.InvokeAsync(Object instance, Object[] arguments) in C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionInvoker.cs:line 52\r\n   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.InvokeWithTimeoutAsync(IFunctionInvoker invoker, ParameterHelper parameterHelper, CancellationTokenSource timeoutTokenSource, CancellationTokenSource functionCancellationTokenSource, Boolean throwOnTimeout, TimeSpan timerInterval, IFunctionInstance instance) in C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs:line 555\r\n   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithWatchersAsync(IFunctionInstanceEx instance, ParameterHelper parameterHelper, ILogger logger, CancellationTokenSource functionCancellationTokenSource) in C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs:line 501\r\n   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, FunctionStartedMessage message, FunctionInstanceLogEntry instanceLogEntry, ParameterHelper parameterHelper, ILogger logger, CancellationToken cancellationToken) in C:\\projects\\azure-webjobs-sdk-rqm4t\\src\\Microsoft.Azure.WebJobs.Host\\Executors\\FunctionExecutor.cs:line 279","Message":"Rate limit for '/acme' reached","Data":{"$type":"System.Collections.ListDictionaryInternal, System.Private.CoreLib"},"InnerException":null,"HelpLink":null,"Source":"ACMESharp","HResult":-2146233088}". See the function execution logs for additional details.". See the function execution logs for additional details.`

**Reply by shibayan** [reply]
I'll also check on this rate limit as it seems odd and has occurred elsewhere as well.

>The “/directory” endpoint and the “/acme” directory & subdirectories have an Overall Requests limit of 40 requests per second.
https://letsencrypt.org/docs/rate-limits/

**Reply by GeorgDangl** [reply]
Same as in this issue: https://github.com/shibayan/keyvault-acmebot/issues/300

**Reply by shibayan** [reply]
This problem has been fixed in the version I released today.
